src/commands/Quotes/addQuote.ts

```diff
@@ -0,0 +1,142 @@
+
+    if (!settingsData.quotesEnabled) {
+      const embed = new BediEmbed()
+          .setColor(colors.ERROR)
+          .setTitle('Add Quote Reply')
+          .setDescription('Quotes are not enabled on this server!');
+      return message.reply({embeds: [embed]});
+    }
+
+    if (settingsData.verificationEnabled && !(await userVerifiedInGuild(author.id, guildId as string))) {
+      const embed = new BediEmbed()
+          .setColor(colors.ERROR)
+          .setTitle('Add Quote Reply')
+          .setDescription(`You are not verified on this server! Run ${surroundStringWithBackTick(settingsData.prefix + 'verify <emailAddress>')}`);
+      return message.reply({embeds: [embed]});
+    }
+    let quote: string | Result<string, UserError>;
+    let quoteAuthor: Result<string, UserError>;
+
+    if (message.reference) {
+      quote = (await message.channel.messages.fetch(message.reference.messageId as Snowflake)).content;
+      quoteAuthor = await args.pickResult('string');
+
+      if (!quoteAuthor.success) return invalidSyntaxReply(message, settingsData);
+    } else {
+
+      quote = await args.pickResult('string');
+      quoteAuthor = await args.pickResult('string');
+
+      if (!quote.success || !quoteAuthor.success) return invalidSyntaxReply(message, settingsData);
+      quote = quote.value;
+    }
+
+    if (quote.length > EMBED_FIELD_MAX_CHAR_LENGTH) {
+      const embed = new BediEmbed()
+          .setColor(colors.ERROR)
+          .setTitle('Add Quote Reply')
+          .setDescription('Quote is too long! Please submit a quote that is 1024 characters or fewer.');
+      return message.reply({embeds: [embed]});
+    }
+
+    const embed = new BediEmbed()
+        .setTitle(`Add Quote Reply - Approvals: 0/${settingsData.quoteApprovalsRequired}`)
+        .setDescription(`Quote: ${surroundStringWithBackTick(quote)}
+        Author: ${surroundStringWithBackTick(quoteAuthor.value as string)}
+        Submitted By: ${author}
+        Approved By:`);
+
+    const row = new MessageActionRow()
+        .addComponents(
+            new MessageButton()
+                .setCustomId('QuoteApprove')
+                .setLabel('Approve')
+                .setStyle('SUCCESS'),
+        );
+
+    const response = await message.reply({
+      embeds: [embed],
+      components: [row],
+    });
+  }
+
+  async onLoad() {
+    client.on('interactionCreate', async interaction => {
+      if (!interaction.isButton() || !(interaction.customId === 'QuoteApprove')) return;
+      await interaction.deferUpdate();
+
+      const message = interaction.message;
+      if (!(message instanceof Message)) return;
+
+      let description = message.embeds[0].description;
+      if (description?.includes(interaction.user.toString())) return;
+
+      const title = message.embeds[0].title;
+      let numApprovals = parseInt(title?.substring('Add Quote Reply - Approvals: '.length, title.indexOf('/')) as string, 10);
```

Comment:
The 'Add Quote Reply - Approvals:` should be coded in somewhere so that if we change one part later it won't look weird and we won't have to update it twice